src/test/java/uk/gov/hmcts/reform/logging/layout/ReformLoggingLayoutTest.java

```diff
@@ -0,0 +1,141 @@
+        DummyP3Exception(String message) {
+            super(AlertLevel.P3, message);
+        }
+    }
+
+    @Parameterized.Parameters(name = "Parsing logback configuration \"{0}\"")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][] {
+            { "logback-test-enable-thread.xml", true, null, null, true },
+            { "logback-test-enable-thread.xml", true, null, AlertLevel.P2, true},
+            { "logback-test-enable-thread.xml", true, null, AlertLevel.P4, true },// will use some other ex
+            { "logback-test-custom-date-format.xml", false, "\\d{2}-\\d{2}-\\d{4}", null, true},
+            { "logback-test-disable-alert-level.xml", false, null, AlertLevel.P3, false }
+        });
+    }
+
+    public ReformLoggingLayoutTest(String resource,
+                                   boolean withThread,
+                                   String dateFormat,
+                                   AlertLevel priority,
+                                   boolean withAlertLevel) throws JoranException, IOException {
+        LoggerContext loggerContext = (LoggerContext) LoggerFactory.getILoggerFactory();
+        JoranConfigurator configurator = new JoranConfigurator();
+
+        InputStream configStream = getClass().getClassLoader().getResourceAsStream(resource);
+        configurator.setContext(loggerContext);
+        configurator.doConfigure(configStream);
+        configStream.close();
+
+        // capture console
+
+        baos = new ByteArrayOutputStream();
+        PrintStream ps = new PrintStream(baos);
+        System.setOut(ps);
+
+        // set logger parameters
+
+        this.withThread = withThread;
+        this.dateFormat = dateFormat;
+        this.priority = priority;
+        this.withAlertLevel = withAlertLevel;
+    }
+
+    @After
+    public void resetConsole() {
+        System.out.flush();
+        System.setOut(old);
+    }
+
+    @Test
+    public void testMessagePatterns() {
+        Logger log = LoggerFactory.getLogger(ReformLoggingLayoutTest.class);
+
+        String message = "message";
+        String level = "INFO ";
+
+        if (withAlertLevel && priority != null) {
+            message = "\\[" + priority.name() + "\\] error";
+            level = "ERROR";
+
+            if (priority.equals(AlertLevel.P4)) {
+                message = String.format(
+                    "\\[%s\\] Bad implementation of '%s' in use",
+                    AlertLevel.P1.name(),
+                    InvalidClassException.class.getCanonicalName()
+                );
+
+                log.error("error", new InvalidClassException("oh no"));
+            } else {
+                log.error("error", new DummyP2Exception("oh no"));
+            }
+        } else if (priority != null) {
+            level = "ERROR";
+
+            log.error(message, new DummyP3Exception("oh no"));
+        } else {
+            log.info(message);
+        }
```

Comment:
I think it would be much easier to understand what's going on with separate tests for each scenario instead of parameterised tests with such an amount of if-else branching